internal/types/path.go

```diff
@@ -17,17 +17,68 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+type Path struct {
+	s []string
+}
+
+// Len returns path length.
+func (p Path) Len() int {
+	return len(p.s)
+}
+
+// Slice returns path slice copy.
+func (p Path) Slice() []string {
+	path := make([]string, len(p.s))
+	copy(path, p.s)
+	return path
+}
+
+// Element returns path element specified by the index or empty string if index below 0 or greater than Path len.
+func (p Path) Element(index int) string {
+	if index < 0 || index > len(p.s) {
```

Comment:
What if `index == len(p.s)`? Wouldn't `p.s[index]` panic?

Reply:
Replaced this method with Suffix/Prefix.